src/core/qgsexpression.cpp

```diff
@@ -3691,6 +3744,8 @@ const QList<QgsExpression::Function*>& QgsExpression::Functions()
     << new StaticFunction( QStringLiteral( "array_remove_all" ), ParameterList() << Parameter( QStringLiteral( "array" ) ) << Parameter( QStringLiteral( "value" ) ), fcnArrayRemoveAll, QStringLiteral( "Arrays" ) )
     << new StaticFunction( QStringLiteral( "array_cat" ), -1, fcnArrayCat, QStringLiteral( "Arrays" ) )
     << new StaticFunction( QStringLiteral( "array_intersect" ), ParameterList() << Parameter( QStringLiteral( "array1" ) ) << Parameter( QStringLiteral( "array2" ) ), fcnArrayIntersect, QStringLiteral( "Arrays" ) )
+    << new StaticFunction( QStringLiteral( "array_to_string" ), -1, fcnArrayToString, QStringLiteral( "Arrays" ) )
+    << new StaticFunction( QStringLiteral( "string_to_array" ), -1, fcnStringToArray, QStringLiteral( "Arrays" ) )
```

Comment:
If you use the ParameterList argument here then all the enforcement of the correct number of arguments and default value will be handled for you. Again, look at the single_sided_buffer function for an example.